src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.cpp

```diff
@@ -544,30 +629,113 @@ QString PCRPrimerDesignForDNAAssemblyTask::getPairReport(U2Region forward,
                                                          const QString &primerName) const {
     QString report;
     if (!forward.isEmpty()) {
-        report += tr("<h2>%1 Forward:</h2>").arg(primerName);
+        report += getPrimerReportTitle(primerName, true);
 
         QString error = tr("Invalid region %1 for %2 forward sequence of length %3").
             arg(forward.toString(), primerName).arg(sequence.length());
         SAFE_POINT(forward.startPos > 0 && forward.startPos < sequence.length() && forward.length > 0, error,
             report += tr("<h3>Error, see log.</h3>"))
 
-        report += QString("<div class=\"seq\"><u>%1</u><b>%2</b></div>").arg(QString(backboneSequence)).
-            arg(QString(sequence.mid(forward.startPos, forward.length)));
+        report += getForwardPrimerReportResult(backboneSequence, sequence.mid(forward.startPos, forward.length));
     }
     if (!reverse.isEmpty()) {
-        report += tr("<h2>%1 Reverse:</h2>").arg(primerName);
+        report += getPrimerReportTitle(primerName, false);
 
         QString error = tr("Invalid region %1 for %2 reverse sequence of length %3").
             arg(reverse.toString(), primerName).arg(sequence.length());
         SAFE_POINT(reverse.startPos > 0 && reverse.startPos < sequence.length() && reverse.length > 0, error,
             report += tr("<h3>Error, see log.</h3>"))
 
-        report += QString("<div class=\"seq\"><b>%1</b><u>%2</u></div>").
-            arg(QString(sequence.mid(reverse.startPos, reverse.length))).arg(QString(backboneSequence));
+        report += getReversePrimerReportResult(backboneSequence, sequence.mid(reverse.startPos, reverse.length));
     }
     return report;
 }
 
+QString PCRPrimerDesignForDNAAssemblyTask::getPairReportForUserPrimers() const {
+    QString report_;
+    if (!settings.forwardUserPrimer.isEmpty() && !settings.reverseUserPrimer.isEmpty()) {
+        report_ += getPrimerReportTitle("C", true);
+        report_ += getForwardPrimerReportResult(backboneSequence, settings.forwardUserPrimer);
```

Comment:
Maybe I didn't mention it, but the backbone sequence shouldn't be added to user primers